app/nginx-1.11.10/src/event/modules/ngx_ff_module.c

```diff
@@ -174,249 +222,216 @@ socket(int domain, int type, int protocol)
         return SYSCALL(socket)(domain, type, protocol);
     }
 
-    return ff_socket(domain, type, protocol);
+    sock = ff_socket(domain, type, protocol);
+
+    if (sock != -1) {
+        sock = convert_fstack_fd(sock);
+    }
+
+    return sock;
 }
 
 int
 bind(int sockfd, const struct sockaddr *addr, socklen_t addrlen)
 {
-    if (unlikely(inited == 0)) {
```

Comment:
This shouldn't be removed.
Because functions defined with `__attribute__((constructor))` may call these functions before nginx is initialized, at this time, `ngx_max_sockets` is not initialized, its value is undefined. So this will cause an error.

Reply:
I've never seen functions defined with ```__attribute__((constructor))``` in nginx and which do call these function. 
Maybe this should be a __coding convention and rule__.

Reply:
Some dynamic libraries may have, and will be called before `main`.
For example, `libcrypto` may call `read/write/close` before `main` on some systems, these symbols may be found at this module. 
See this 10c5711ed22017a7b558c018e2ee92c2ecd85547.

Reply:
I've fixed it.